chain/xrp/tx/tx.go

```diff
@@ -0,0 +1,229 @@
+package tx
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	xc "github.com/cordialsys/crosschain"
+	btctx "github.com/cordialsys/crosschain/chain/bitcoin/tx"
+	binarycodec "github.com/xyield/xrpl-go/binary-codec"
+	"strings"
+)
+
+const (
+	PAYMENT                 TransactionType = "Payment"
+	TRANSACTION_HASH_PREFIX                 = "54584E00"
+)
+
+type TransactionType string
+
+type XRPTransaction struct {
+	Account            xc.Address       `json:"Account"`
+	Amount             AmountBlockchain `json:"Amount"`
+	Destination        xc.Address       `json:"Destination"`
+	Fee                string           `json:"Fee"`
+	Flags              int64            `json:"Flags,omitempty"`
+	LastLedgerSequence int64            `json:"LastLedgerSequence"`
+	Sequence           int64            `json:"Sequence"`
+	SigningPubKey      string           `json:"SigningPubKey"`
+	TransactionType    TransactionType  `json:"TransactionType"`
+	TxnSignature       string           `json:"TxnSignature"`
+}
+
+type AmountBlockchain struct {
+	StringValue string  `json:"-"`
+	AmountValue *Amount `json:"-"`
+	IsString    bool    `json:"-"`
+}
+
+type Amount struct {
+	Currency string `json:"currency"`
+	Issuer   string `json:"issuer"`
+	Value    string `json:"value"`
+}
```

Comment:
I would rewrite this as 

```go
// XRP amount is either string or Token Amount
type XRPOrTokenAmount struct {
	XRPAmount string  `json:"-"`
	TokenAmount *Amount `json:"-"`
}

type TokenAmount struct {
	Currency string `json:"currency"`
	Issuer   string `json:"issuer"`
	Value    string `json:"value"`
}
```

Don't need `.IsString` bool, can just check if `XRPAmount` is empty string or not.